rhp/v4/validation.go

```diff
@@ -0,0 +1,127 @@
+package rhp
+
+import (
+	"errors"
+	"fmt"
+
+	"go.sia.tech/core/types"
+)
+
+// Validate validates a read sector request.
+func (req *RPCReadSectorRequest) Validate() error {
+	switch {
+	case req.Length == 0:
+		return errors.New("length must be greater than 0")
+	case req.Offset+req.Length > SectorSize:
+		return errors.New("read request exceeds sector bounds")
+	case (req.Offset+req.Length)%LeafSize != 0:
```

Comment:
did we consider making `Offset` and `Length` multiples of `LeafSize`? (i.e. rename them to `LeafIndex` and `NumLeaves`?)